app/plugin_signature.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+)
+
+// VerifyPlugin checks that the given signature corresponds to the given plugin and matches a trusted certificate.
+func (a *App) VerifyPlugin(plugin, signature io.Reader) *model.AppError {
+	if verifySignature(bytes.NewReader(mattermostPublicKey), plugin, signature) == nil {
```

Comment:
this is equivalent of doing 
`if err := verifySignature(bytes.NewReader(mattermostPublicKey), plugin, signature); err == nil`

But I feel like storing the `err` is clearer as you know what you're expecting in return as opposed to just checking the return value, as there is an ambiguity there as to what might be the return value, is it an error or maybe a slice or pointer or anything else. Plus it's also more go idiomatic.